testsuite/integration/web/src/test/java/org/jboss/as/test/integration/web/cookie/DefaultCookieVersionTestCase.java

```diff
@@ -21,83 +21,95 @@
  *
- * @author prabhat.jha@jboss.com
- * @author lbarreiro@redhat.com
+ * @author Stuart Douglas
+ * @author Jan Stourac
  */
 @RunWith(Arquillian.class)
 @RunAsClient
-@ServerSetup(DefaultCookieVersionTestCase.CookieVersionServerSetup.class)
 public class DefaultCookieVersionTestCase {
     @ArquillianResource(CookieServlet.class)
     protected URL cookieURL;
 
-    static class CookieVersionServerSetup implements ServerSetupTask {
+    private static String DEF_SERVLET_ADDR = "/subsystem=undertow/servlet-container=default";
 
-        @Override
-        public void setup(ManagementClient managementClient, String s) throws Exception {
-            ModelNode modelNode = new ModelNode();
-            modelNode.get(ModelDescriptionConstants.OP).set(ModelDescriptionConstants.WRITE_ATTRIBUTE_OPERATION);
-            modelNode.get(ModelDescriptionConstants.OP_ADDR).set(PathAddress.parseCLIStyleAddress("/subsystem=undertow/servlet-container=default").toModelNode());
-            modelNode.get(ModelDescriptionConstants.NAME).set("default-cookie-version");
-            modelNode.get(ModelDescriptionConstants.VALUE).set(1);
-            ModelNode reuslt = managementClient.getControllerClient().execute(modelNode);
-            ServerReload.executeReloadAndWaitForCompletion(managementClient);
-        }
-
-        @Override
-        public void tearDown(ManagementClient managementClient, String s) throws Exception {
-
-            ModelNode modelNode = new ModelNode();
-            modelNode.get(ModelDescriptionConstants.OP).set(ModelDescriptionConstants.UNDEFINE_ATTRIBUTE_OPERATION);
-            modelNode.get(ModelDescriptionConstants.OP_ADDR).set(PathAddress.parseCLIStyleAddress("/subsystem=undertow/servlet-container=default").toModelNode());
-            modelNode.get(ModelDescriptionConstants.NAME).set("default-cookie-version");
-            ModelNode reuslt = managementClient.getControllerClient().execute(modelNode);
-            ServerReload.executeReloadAndWaitForCompletion(managementClient);
-        }
+    @AfterClass
+    public static void tearDown() throws IOException {
+        configureDefaultCookieVersion(null);
     }
 
     @Deployment
     public static WebArchive deployment() {
 
         WebArchive war = ShrinkWrap.create(WebArchive.class, "jbosstest-cookie.war");
-        war.addClass(CookieReadServlet.class);
         war.addClass(CookieServlet.class);
+        war.addClass(CookieEchoServlet.class);
 
         return war;
     }
 
+    @ContainerResource
+    private static ManagementClient managementClient;
+
+    @Test
+    public void testDefaultCookieVersion0() throws Exception {
+        commonDefaultCookieVersion(0);
+    }
+
     @Test
-    public void testDefaultCookieVersion() throws Exception {
+    public void testDefaultCookieVersion1() throws Exception {
+        commonDefaultCookieVersion(1);
+    }
+
+    @Test
+    public void testSendCookieVersion0() throws Exception {
+        commonSendCookieVersion(0);
```

Comment:
If you want to test version 0 you will need to use a different servlet than the existing one. There are some circumstances where version 0 cookies are upgraded to v1, and the servlet that I re-used for this test contains some of them.

Reply:
OK, I understand now... e.g. in case that cookie has a comment... Is it ok for you if I use different servlet for all tests in this testsuite where just simple cookies are created? So we actually know that only the `default-cookie-version` attribute has been involved and no internal Undertow cookie upgrade mechanism.